app/controllers/api/base_controller.rb

```diff
@@ -38,17 +38,20 @@ def get_resource
 
     # overwrites resource_scope in FindCommon to consider nested objects
     def resource_scope(options = {})
-      options[:association] ||= controller_name
-      if nested_obj && nested_obj.respond_to?(options[:association])
-        association = nested_obj.send(options[:association])
-        if association.respond_to?(:authorized)
-          association.authorized(options[:permission], resource_class)
-        else
-          association
-        end
-      else
-        super(options)
-      end
+      child_scope = super(options)
+      child_scope.merge(parent_scope).readonly(false)
+    end
+
+    def parent_scope(options = {})
+      parent_name, parent_resource_scope = parent_resource_details()
+
+      return resource_class.where(nil) unless parent_resource_scope
```

Comment:
It's not a hack, this is the way [suggested by the community] (http://stackoverflow.com/questions/18198963/with-rails-4-model-scoped-is-deprecated-but-model-all-cant-replace-it) to get a relation object in rails 4.
And it's not a warning - it just breaks the code. If I have two ways doing the same thing, I will prefer the one that will last longer.